### ISSUE TITLE
Apicurio editor lags with large number of paths/data types in list

### ISSUE THREAD

**Issue by colvinco** [Apicurio/apicurio-studio#1815]
I didn't find an open issue for this, sorry if I missed it. https://github.com/Apicurio/apicurio-studio/issues/791#issuecomment-503558047 touches on the issue but it's closed - it's still a problem though.

We have a spec with >100 paths and >350 data types. With the full list of types present, every action in the editor has about a 2-3 second delay. e.g. clicking on a dropdown, editing a description, etc
Collapsing the list has no effect, but if I filter the path/data types list to have ~250 entries in total then it is reasonably responsive. If I filter it more then it's totally responsive (~150 entries seems to be fine).

**Comment by EricWittmann** [comment]
Yeah this is an ongoing fight I have with Angular.  I've actually done quite a lot to try and speed things up, and it's still not very good with large data models.  Frankly I'm not sure how much more I can improve it.  Angular is very aggressive in how it tries to detect model changes so that it can selectively update UI elements.  Every user action results in a reconciliation step where angular compares what's being displayed to the latest data model.  Even if I know the data model hasn't changed, it's hard to prevent angular from doing a lot of work on the comparison.

I'm sure there must be something additional that can be done, but am not quite sure what it is yet.

**Comment by branch-whisperer** [comment]
I'm pretty inexperienced with Angular so I could be wrong here, but I was looking at the pages' components and it seems they are using the default [ChangeDetectionStrategy](https://angular.io/api/core/ChangeDetectionStrategy), which means any time anything changes, Angular rechecks everything. If that's the case configuring components to only be updated when changes are pushed or an update is manually triggered should be a huge performance boost. [This video](https://www.youtube.com/watch?v=-tB-QDrPmuI) explains it better than I could (I have no relation with the video).

**Comment by EricWittmann** [comment]
Honestly the solution to this will likely ultimately be a re-implementation using React.... :)
